### ISSUE TITLE
Optimize Model Performance

### ISSUE THREAD

**Issue by dharanee99** [dharanee99/DL-FinalProject#10]
Fine-tune model depth, architecture, and hyperparameters for improved accuracy and generalization.
